src/cpp/session/modules/SessionFind.cpp

```diff
@@ -1060,12 +1086,33 @@ core::Error runGrepOperation(const GrepOptions& grepOptions, const ReplaceOption
       cmd << "--include=" + filePattern.getString();
    }
 
-   cmd << shell_utils::EscapeFilesOnly << "--" << shell_utils::EscapeAll;
-
-
    // Filepaths received from the client will be UTF-8 encoded;
    // convert to system encoding here.
    FilePath dirPath = module_context::resolveAliasedPath(grepOptions.directory);
+   for (json::Value filePattern : grepOptions.excludeFilePatterns)
+   {
+      // to retrieve the git exclusions, we run a git command and parse the output
+      if (filePattern.getString().compare("gitExclusions") == 0)
```

Comment:
`getString()` can throw when the `Value` is not a string, so for safety would recommend testing and skipping non-string values in the array (realize this is unlikely but the consequence of an exception here is a full session crash!)

Reply:
Fixed